woocommerce/woocommerce-android
# Products parent-child relationship

Closes: #8403

### Why
On the Eagle team, we are working on improving the experience for our users across our Canonical Extensions. One of these extensions is Bundled Products, and one of our tasks and ideas to improve the experience was to display the relationship between a bundle and its products.

### Description
This PR displays the parent-child relationship between a bundled product and its child products by adding a new type of product: the grouped product, a product with a list of child products. Now, when receiving the list of products on the order detail screen view model, we process the list by the product's relationship. In the case of bundled products, it displays the bundle product as the parent and all products inside the bundle as its children.

⚠️ This PR only contains the visual changes needed to display the parent-child relationship, to achieve this goal we have forced some changes in the `toOrderProducts`.

### Testing instructions
1. Open the orders tab
2. Select any order
3. Check that children products appears with more margin than parent/simple products 

### Images/gif

https://github.com/woocommerce/woocommerce-android/assets/18119390/07b853e1-0268-489a-bc48-6e4eff2262bf
